### PR TITLE
Use iam_units.convert_gwp

### PR DESCRIPTION
**Pull request by khaeru**
Adjust to use the code from IAMconsortium/units#12.

**PR checklist**

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md added

### CODE REVIEW

File: doc/source/tutorials/unit_conversion.ipynb [hunk 1]
```diff
@@ -221,7 +221,7 @@
     "gwp = 'AR5GWP100'\n",
     "target = 'Mt CO2e/yr'\n",
     "(\n",
-    "    df.convert_unit('Mt CH4/yr', to=target, context=f'gwp_{gwp}')\n",
+    "    df.convert_unit('Mt CH4/yr', to=target, context=gwp)\n",
```
Comment (danielhuppmann) [inline doc/source/tutorials/unit_conversion.ipynb:224]: please keep `gwp_*` naming convention

File: doc/source/tutorials/unit_conversion.ipynb [hunk 2]
```diff
@@ -195,7 +195,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "df.convert_unit('Mt CH4/yr', to='Mt CO2e/yr', context='gwp_AR5GWP100').timeseries()"
+    "df.convert_unit('Mt CH4/yr', to='Mt CO2e/yr', context='AR5GWP100').timeseries()"
```
Comment (danielhuppmann) [inline doc/source/tutorials/unit_conversion.ipynb:198]: please keep `gwp_*` naming convention

File: pyam/core.py [hunk 3]
```diff
@@ -766,41 +766,72 @@ def rename(self, mapping=None, inplace=False, append=False,
...
+        `iam-units <https://github.com/IAMconsortium/units>`_ package.
+        This registry can also be accessed directly, using::
+
+            from iam_units import registry
+
+        Using the :mod:`iam_units` *registry*, *current* and *to* may contain
+        the names of greenhouse gas (GHG) species, such as 'CO2e', 'C', 'CH4',
+        'N2O', 'HFC236fa', etc. In this case, *context* must contain the name
+        of a specific global warming potential (GWP) metric supported by
+        :mod:`iam_units`, such as 'AR5GWP100'.
```
Comment (danielhuppmann) [inline pyam/core.py:791]: please keep `gwp_*` naming convention

File: pyam/core.py [hunk 4]
```diff
@@ -766,41 +766,72 @@ def rename(self, mapping=None, inplace=False, append=False,
...
+            if attempting a GWP conversion but *context* is not given.
+        pint.DimensionalityError
+            without *factor*, when *current* and *to* are not compatible units.
         """
         # TODO: deprecate using `dict` in next release (>=0.6.0)
         # TODO: make `to` required
+
+        # Handle user input
+        # Check that (only) either factor or registry/context is provided
+        if factor and any([registry, context]):
```
[the remark this answers is not in the document]
Comment (khaeru) [reply pyam/core.py:832]: I was following the split used in ixmp, i.e.
- core.py, the user API definition handles/sanitizes user input: checking that arguments are correct where possible, transforming shorthands, etc., and then…
- passes on to 'internal' code with simpler signatures.

I wasn't sure of the approach in pyam—I can move it back if the approach is to contain everything in units.py

[the remark this answers is not in the document]
Comment (danielhuppmann) [reply pyam/core.py:832]: got it, don't think that we have a well-defined approach here...

**Comment by danielhuppmann** [comment]
@khaeru can you please add a line to the release notes that `iam-units` was refactored from a submodule to a dependency?

**Comment by khaeru** [comment]
> @khaeru can you please add a line to the release notes that `iam-units` was refactored from a submodule to a dependency?

Done!

**Pull request merged**